### PR TITLE
[nginx-ingress-controller] Add ssl certificate checksum to template

### LINKED ISSUE #800: Nginx Ingress Controller - Config file not updated when a secret is updated
It seems that updating a secret does not trigger the config file to be updated. I have a script that is automating let's encrypt. But after the certificate is fetched and the Secret resource updated, the nginx configuration is not rebuilt (so hosts waiting on the certificate are updated). Is this intentional?

### PR DESCRIPTION
**Pull request by aledbf**
fixes #800

### CODE REVIEW

**Comment by simonswine** [comment]
cherry-picked with #1063 and works perfectly in my cluster

LGTM

File: ingress/controllers/nginx/nginx.tmpl [hunk 1]
```diff
@@ -155,6 +155,7 @@ http {
     server {
         listen 80{{ if $cfg.useProxyProtocol }} proxy_protocol{{ end }};
         {{ if $server.SSL }}listen 443{{ if $cfg.useProxyProtocol }} proxy_protocol{{ end }} ssl http2;
+        # PEM sha: {{ $server.SSLPemChecksum }}
```
Comment (bprashanth) [inline ingress/controllers/nginx/nginx.tmpl]: why comment this out?

File: ingress/controllers/nginx/nginx/ssl.go [hunk 2]
```diff
@@ -26,28 +28,52 @@ import (
...
+		PemFileName:  pemFileName,
+		PemSHA:       nginx.pemSHA1(pemFileName),
+		CN:           cn,
+	}, nil
 }
 
 // CheckSSLCertificate checks if the certificate and key file are valid
 // returning the result of the validation and the list of hostnames
 // contained in the common name/s
-func (nginx *Manager) CheckSSLCertificate(pemFileName string) ([]string, error) {
```
Comment (bprashanth) [inline ingress/controllers/nginx/nginx/ssl.go]: please fix comment to match method name

File: ingress/controllers/nginx/nginx.tmpl (hunk 1 above)
Comment (aledbf) [inline ingress/controllers/nginx/nginx.tmpl]: Is just a comment. If the secret changes this comment will trigger the reload (diff with the current running cfg)

File: ingress/controllers/nginx/nginx/ssl.go (hunk 2 above)
Comment (aledbf) [inline ingress/controllers/nginx/nginx/ssl.go]: done

File: ingress/controllers/nginx/nginx.tmpl (hunk 1 above)
Comment (bprashanth) [inline ingress/controllers/nginx/nginx.tmpl]: oh, hmm. suggest adding a commet about the significance of that comment.

Comment (aledbf) [inline ingress/controllers/nginx/nginx.tmpl]: done

**Comment by bprashanth** [comment]
Thanks, LGTM

**Pull request merged**